modules/logging/filters_test.go

```diff
@@ -0,0 +1,19 @@
+package logging
+
+import (
+	"testing"
+
+	"go.uber.org/zap/zapcore"
+)
+
+func TestQueryFilter(t *testing.T) {
+	f := QueryFilter{[]queryFilterAction{
+		{ReplaceType, "foo", "REDACTED"},
+		{DeleteType, "bar", ""},
+	}}
+
+	out := f.Filter(zapcore.Field{String: "/path?foo=a&foo=b&bar=c&bar=d&baz=e"})
+	if out.String != "/path?baz=e&foo=REDACTED" {
```

Comment:
To me it's not really intuitive (despite the obvious goal to redact the values) that two `foo` keys go in (values `a` and `b`) but only one `foo` key is expected to come out. Is this somehow rooted in the way caddy in general treats queries and logs them?

Reply:
It's rooted in the Go stdlib but maybe can we preserve the same number of keys, I'll take a look. As it's on the hot path, we should be careful to not consume too much resources.

Reply:
Actually, it was easy to handle. Done, thanks @wiese!